### PR TITLE
[OTel] Test `analytics.event` override with respect to OTLP Code

### PR DESCRIPTION
**Pull request by PROFeNoM**
## Motivation

The [RFC](https://docs.google.com/document/d/1KB6XQxZDUgSSQHp1TEk_0X6zgOzmB9-r8NKE9TVaRJw/edit#bookmark=id.44nna34oqd3j) states that the reserved span attributes must match the behavior of the [OTLP code](https://github.com/DataDog/datadog-agent/blob/ea1911d5618df80c1dd25b362369a2e0491910be/pkg/trace/api/otlp.go#L466-L483) which uses Go's [strconv.ParseBool](https://cs.opensource.google/go/go/+/refs/tags/go1.21.5:src/strconv/atob.go;l=10).
```
// ParseBool returns the boolean value represented by the string.
// It accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False.
```

More specifically, this means that 1, t, T, 0, f, and F strings are also valid entries, while anything other than those listed above shouldn't set the `_dd1.sr.eausr` in the metrics. This PR tests this behavior.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

**Review by PerfectSlayer — COMMENTED** [review]
I left a comment on Slack about it :)

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
CI is failing

**Review by cbeauchesne — APPROVED** [review]
So far, ok for me. But I suggest getting a review from another eyes familiar with the RFC to validate the test logic.

**Comment by PROFeNoM** [comment]
Added apm Go/Java/Ruby/Dotnet teams as reviewers since they will be directly impacted by this change. :+1:

File: tests/parametric/test_otel_span_methods.py [hunk 1]
```diff
@@ -483,9 +483,10 @@ def test_otel_span_reserved_attributes_overrides(self, test_agent, test_library)
         assert "span.type" not in span["meta"]
         assert "analytics.event" not in span["meta"]
 
-    @missing_feature(context.library < "java@1.25.0", reason="Implemented in 1.25.0")
+    @missing_feature(context.library == "java", reason="OTLP behavior not respected")
```
Comment (PerfectSlayer) [inline tests/parametric/test_otel_span_methods.py:486]: This is not related to OTLP. We choose to not follow the golang boolean string parsing convention.

An honest reason would be "choose to not implement go parsing logic".

Comment (PROFeNoM) [reply tests/parametric/test_otel_span_methods.py:486]: Ok, will change this :+1:

Comment (PerfectSlayer) [reply tests/parametric/test_otel_span_methods.py]: What about making a second test case with the extended parameters? So we can keep testing the expected behavior on the usual boolean (including using Java) and have a test covering the extended values.
As the tests is parametrize, it should be easy.

Comment (PROFeNoM) [reply tests/parametric/test_otel_span_methods.py]: Makes sense. I'd obviously prefer that we all have the same behavior, but yeah.

Comment (PerfectSlayer) [reply tests/parametric/test_otel_span_methods.py]: In the end, it will still result of having a partial support feature dashboard for Java so it won't be unnoticed. 
Instead, you will just have a more granular evaluation.

**Review by bouwkast — COMMENTED** [review]
I think that it follows the RFC and OTLP.go better, but I think that it may make sense to ask the overall team whether it makes sense to change the RFC to reduce the overall scope to require the simpler boolean logic that was defined previously and then _optionally_ support the more extensive `strconv.ParseBool` if it makes sense for your language. (I'm assuming that there may be performance concerns with this way?)

The main issue with .NET previously was whether `something-else` should _add_ and _set_ the `_dd1.sr.eausr` tag. From a behavior perspective the lack of the tag or it being set to `0` will be equivalent - so I'm thinking this is more of a semantic question.

Maybe further splitting the tests to define three tests in total:

- One that covers the simpler logic of `"false"`, `"False"`, '"true"` (basically previous test without `"something-else"`
- One that covers _how_ random string data is handled (e.g., `"something-else"` either setting the tag to `0`  or not setting the tag at all
- One that covers the more extended parsing logic as it is currently

Maybe this is taking it too far for this?

File: tests/parametric/test_otel_span_methods.py [hunk 2]
```diff
@@ -541,5 +563,8 @@ def run_otel_span_reserved_attributes_overrides_analytics_event(
     assert len(trace) == 1
 
     span = get_span(test_agent)
-    assert span["metrics"].get("_dd1.sr.eausr") == expected_metric_value
+    if (expected_metric_value is not None) or (context.library not in ["dotnet"]):
```
Comment (ajgajg1134) [inline tests/parametric/test_otel_span_methods.py:566]: Hmm why are we checking for library here and not using the `missing_feature` decorator?

File: tests/parametric/test_otel_span_methods.py (hunk 1 above)
Comment (marcotc) [reply tests/parametric/test_otel_span_methods.py]: >We choose to not follow the golang boolean string parsing convention.

Same for Ruby, and I see that you have already updated the skip reason description @PROFeNoM so thank you!

I also don't foresee us making this test pass for the provided values in Ruby, given that:
* This specific boolean parsing is not consistent with other boolean environment variables at Datadog
* The full list of values that `strconv#ParseBool` considers true and false look pretty foreign to the Ruby environment.

File: tests/parametric/test_otel_span_methods.py [hunk 3]
```diff
@@ -484,29 +484,63 @@ def test_otel_span_reserved_attributes_overrides(self, test_agent, test_library)
...
+            for inputs that aren't accepted by strconv.ParseBool
+        """
+        run_otel_span_reserved_attributes_overrides_analytics_event(
+            analytics_event_value=analytics_event_value,
+            expected_metric_value=expected_metric_value,
+            test_library=test_library,
+            test_agent=test_agent,
+        )
+
+    @missing_feature(context.library == "java", reason="Choose to not implement Go parsing logic")
```
Comment (cbeauchesne) [inline tests/parametric/test_otel_span_methods.py:531]: If the reason is really an educated choice, then you should decorate the test with `@irrelevant`, so it won't show up as not. implemented in the feature parity dashbaord.

Comment (PROFeNoM) [reply tests/parametric/test_otel_span_methods.py]: Thanks, I didn't know! Changed it 😃

File: tests/parametric/test_otel_span_methods.py [hunk 4]
```diff
@@ -484,29 +484,63 @@ def test_otel_span_reserved_attributes_overrides(self, test_agent, test_library)
...
+            Tests the analytics.event reserved attribute override with basic inputs
+        """
+        run_otel_span_reserved_attributes_overrides_analytics_event(
+            analytics_event_value=analytics_event_value,
+            expected_metric_value=expected_metric_value,
+            test_library=test_library,
+            test_agent=test_agent,
+        )
+
+    @irrelevant(context.library == "java", reason="The expected metric value is zero")
```
Comment (cbeauchesne) [inline tests/parametric/test_otel_span_methods.py:508]: For this one, the reason does not explain very well why it's irrelevant : 

* does it mean the test is not good because *The expected metric value is zero* -> so we need to update the test
* does it mean that those tracer decided to set the metric to zero -> the reason must make claer it's an educated choice
* the expected metric is really zero, but those tracer are failing ? -> it must be decorated with `@bug`
* other interpretation I may not see ?

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: tests/parametric/test_otel_span_methods.py (hunk 4 above)
Comment (PROFeNoM) [reply tests/parametric/test_otel_span_methods.py:508]: With respect to the RFC, it would be the second one. Here is what comes to my mind:
- `Choose to set the metric value to zero for parseBool-rejected values` 
- `Chose zero metric for parseBool-rejected inputs`
- `Set metric to zero when rejected by parseBool`

Comment (cbeauchesne) [reply tests/parametric/test_otel_span_methods.py:508]: WDYT  about a message making  crystal clear the fact it's a design choice ? Something like  `Java tracer decided to ...`

Comment (PROFeNoM) [reply tests/parametric/test_otel_span_methods.py]: Alright, I changed the message to `<language> tracer decided to always set _dd1.sr.eausr: 1 for truthy analytics.event inputs, else 0", which imo, clearly says what's going on, although it could be argued that "truthy" is vague (and it is).

**Review by cbeauchesne — APPROVED** [review]
AGTM

File: tests/parametric/test_otel_span_methods.py [hunk 5]
```diff
@@ -484,29 +484,72 @@ def test_otel_span_reserved_attributes_overrides(self, test_agent, test_library)
         assert "analytics.event" not in span["meta"]
 
     @missing_feature(context.library < "java@1.25.0", reason="Implemented in 1.25.0")
-    @missing_feature(context.library == "nodejs", reason="Not implemented")
     @missing_feature(context.library == "dotnet", reason=".NET dosn't treat 'something-else' as a valid input here.")
```
Comment (bouwkast) [inline tests/parametric/test_otel_span_methods.py:487]: This can be un-skipped for .NET as `something-else` is no longer in this test and we pass all of them.

Comment (bouwkast) [reply tests/parametric/test_otel_span_methods.py:487]:
```suggestion
```
